src/diffusers/loaders.py

```diff
@@ -67,6 +67,36 @@
 CUSTOM_DIFFUSION_WEIGHT_NAME_SAFE = "pytorch_custom_diffusion_weights.safetensors"
 
 
+class PatchedLoraProjection(nn.Module):
+    def __init__(self, regular_linear_layer, lora_linear_layer, lora_scale=1):
+        super().__init__()
+        self.regular_linear_layer = regular_linear_layer
+        self.lora_linear_layer = lora_linear_layer
+        self.lora_scale = lora_scale
+
+    def forward(self, input):
+        return self.regular_linear_layer(input) + self.lora_scale * self.lora_linear_layer(input)
```

Comment:
Previously we were monkey patching the forward method with a nested function and then storing the old forward method on module as a new instance variable.

This is a bit cleaner because we don't have to add a new instance variable to an object we didn't construct and nested functions should generally be avoided if possible imo - just make the new scope explicit and avoid having to remember the languages scoping rules

Reply:
Agree